elfo-core/src/init.rs

```diff
@@ -29,50 +29,74 @@ use crate::{
 
 type Result<T, E = StartError> = std::result::Result<T, E>;
 
-async fn start_entrypoints(ctx: &Context, topology: &Topology) -> Result<()> {
+async fn start_entrypoints(ctx: &Context, topology: &Topology, is_check_only: bool) -> Result<()> {
     let futures = topology
         .actor_groups()
         .filter(|group| group.is_entrypoint)
-        .map(|group| {
-            let config = Default::default();
-            ctx.request_to(group.addr, UpdateConfig { config })
+        .map(|group| async move {
+            let response = ctx
+                .request_to(
+                    group.addr,
+                    UpdateConfig {
+                        config: Default::default(),
+                    },
+                )
                 .resolve()
-                .or_else(|err| async move {
-                    match err {
-                        RequestError::Ignored => Ok(Ok(())),
-                        _ => Err(StartError::Other(
-                            "initial messages cannot be delivered".into(),
-                        )),
-                    }
-                })
+                .await;
+            match response {
+                Ok(Ok(())) => Ok(()),
+                Ok(Err(e)) => Err(StartError::EntrypointErrors(EntrypointErrors {
+                    errors: vec![EntrypointError {
+                        group: group.name.clone(),
+                        reason: e.reason,
+                    }],
+                })),
+                Err(RequestError::Ignored) => Ok(()),
+                Err(RequestError::Closed(..)) => Err(StartError::Other(
+                    "config cannot be delivered to the entrypoint".into(),
+                )),
+            }?;
```

Comment:
Just realized that we still can extract this into function, just pass the message for `StartError::Other` into it